pkg/etc/init.d/wavefront-proxy

```diff
@@ -106,28 +107,36 @@ download_jre() {
             echo "Authenticating as $proxy_user" >&2
             proxy_args+=" --proxy-user $proxy_user:$proxy_password"
         fi
-    else
-        echo "No HTTP proxy configuration detected - attempting direct download" >&2
     fi
-    curl -L --silent -o /tmp/jre.tar.gz $proxy_args https://s3-us-west-2.amazonaws.com/wavefront-misc/proxy-jre-11.0.6-linux_x64.tar.gz || true
+    echo "Downloading and installing JRE $bundled_jre_version" >&2
+    curl -L --silent -o /tmp/jre.tar.gz $proxy_args https://s3-us-west-2.amazonaws.com/wavefront-misc/proxy-jre-$bundled_jre_version-linux_x64.tar.gz || true
     tar -xf /tmp/jre.tar.gz --strip 1 -C $proxy_jre_dir || true
     rm /tmp/jre.tar.gz || true
 }
 
 # If $PROXY_JAVA_HOME is not defined and there is no JRE in $proxy_jre_dir, try to auto-detect
-# locally installed JDK first. We will accept 1.8, 9, 10, 11.
+# locally installed JDK first. We will accept 8, 9, 10, 11.
 if [[ -z "$PROXY_JAVA_HOME" && ! -r $proxy_jre_dir/bin/java ]]; then
-    if type -p java; then
-        JAVA_VERSION=$(java -version 2>&1 | head -1 | cut -d'"' -f2 | sed '/^1\./s///' | cut -d'.' -f1)
-        if [[ $JAVA_VERSION == "11" || $JAVA_VERSION == "10" || $JAVA_VERSION == "9" || $JAVA_VERSION == "8" || $JAVA_VERSION == "1.8" ]]; then
-            JAVA_HOME=$($(dirname $(dirname $(readlink -f $(which javac)))))
-            echo "Using Java runtime $JAVA_VERSION detected in $JAVA_HOME" >&2
+    # if java found in path
+    if type -p java > /dev/null 2>&1; then
+        JAVA_VERSION=$(java -version 2>&1 | head -1 | cut -d'"' -f2 | sed '/^1\./s///' | cut -d'.' -f1) || true
+        if [[ -z "$JAVA_VERSION" ]]; then
+          JAVA_VERSION="(unknown version)"
+        fi
```

Comment:
why do we need to set it to unknown? it will in any case be installed when the flow goes to line number #134 else block

Reply:
purely for message output reason - "Found Java runtime (unknown version), needs 8, 9, 10 or 11 to run" reads better than "Found Java runtime , needs 8, 9, 10 or 11 to run"